packages/flutter_tools/lib/src/runner/target_devices.dart

```diff
@@ -29,7 +29,8 @@ String _foundSpecifiedDevicesMessage(int count, String deviceId) =>
 String _noMatchingDeviceMessage(String deviceId) => 'No supported devices found with name or id '
     "matching '$deviceId'.";
 String flutterSpecifiedDeviceDevModeDisabled(String deviceName) => 'To use '
-    "'$deviceName' for development, enable Developer Mode in Settings → Privacy & Security.";
+    "'$deviceName' for development, enable Developer Mode in Settings → Privacy & Security. "
+    'You may also need to open XCode while the device is connected, and say yes when prompted to Trust the device.';
```

Comment:
@jmagman I think you don't need XCode to be open to get prompted, right?

Reply:
When I unpair my phone and turn off developer mode and run `flutter devices` or `doctor` I see:
```
Error: Purple 14 is not paired with your computer. To use Purple 14 with Xcode, unlock it and choose to trust this computer when prompted. (code -9)
```
However that doesn't seem to be checked during `flutter run` and I get:
```
Could not run build/ios/iphoneos/Runner.app on iPhone.
Try launching Xcode and selecting "Product > Run" to fix the problem:
  open ios/Runner.xcworkspace

Installing and launching...                                         71ms
Error launching application on iPhone.
```

I can't seem to reproduce the issue where the phone is unpaired and it prints the `enable Developer Mode in Settings → Privacy & Security` error message.  I'm not sure when Xcode would _not_ open during `flutter run` since we now launch it.  An issue with reproduction steps would have been helpful 🙂 

It would be better if the tool checked if it's paired during device discovery, I can't find an issue for that at the moment, filed https://github.com/flutter/flutter/issues/144447

Reply:
PR out at https://github.com/flutter/flutter/pull/144551